apps/emqx_auth/src/emqx_authz/emqx_authz_rule.erl

```diff
@@ -140,7 +141,7 @@ compile(Permission, Who, Action, TopicFilters) ->
 compile({Permission, all}) when
     ?IS_PERMISSION(Permission)
 ->
-    {Permission, all, all, [compile_topic(<<"#">>)]};
+    {Permission, all, all, [?ANY_TOPIC]};
 compile({Permission, Who, Action, TopicFilters}) when
```

Comment:
should `'all'` also be supported in specific rule?
It only supports `topic() | topic_filter() | {eq, topic() | topic_filter()}`.


The existing topicfilter list cannot handle the following usage.
`{allow, {username, "foo"}, publish, all}.`
`{deny, {username, "bar"}, subscrube, all}.`

Reply:
make sense. I'll add it.

Reply:
they should be different in json or raw file.
```json
{
...
  "topics": ["t/1", "foo/bar"], // current supported

  "topics": "all" // means the atom 'all'
...
}
```

Reply:
the problem is, it allows both `topic` and `topics`
I'll try to add a special handling for `topics` to support `all`, but not `topic`.